app/dashboard/static/js/app/view-tests-plan-id.2020.2.js

```diff
@@ -0,0 +1,459 @@
+            document.getElementById('plan-name'),
+            document.createTextNode(results.name));
+        html.replaceContent(
+            document.getElementById('device-type'),
+            document.createTextNode(results.device_type));
+        html.replaceContent(
+            document.getElementById('tree'), treeNode);
+        html.replaceContent(
+            document.getElementById('git-branch'),
+            document.createTextNode(results.git_branch));
+        html.replaceContent(
+            document.getElementById('git-describe'), describeNode)
+        html.replaceContent(
+            document.getElementById('git-url'), gitNode);
+        html.replaceContent(
+            document.getElementById('git-commit'),
+            document.createTextNode(results.git_commit));
+        html.replaceContent(
+            document.getElementById('arch'),
+            document.createTextNode(results.arch));
+        html.replaceContent(
+            document.getElementById('defconfig'),
+            document.createTextNode(results.defconfig_full));
+        html.replaceContent(
+            document.getElementById('compiler'),
+            document.createTextNode(results.compiler_version_full));
+        html.replaceContent(
+            document.getElementById('job-date'), dateNode);
+        html.replaceContent(
+            document.getElementById('job-log'), logNode);
+    }
+
+    function updateChart(testCount) {
+        function countTests(tc) {
+            return [
+                tc['total'],
+                [tc['pass'], tc['regressions'], tc['unknown']]
+            ];
+        }
+
+        pieChart.testpie({
+            element: 'test-chart',
+            countFunc: countTests,
+            response: testCount,
+            legend: true,
+            legendIds: {
+                'pass': '#show-pass',
+                'fail': '#show-fail',
+                'unknown': '#show-unknown',
+            },
+            legendTitles: {
+                'pass': 'Successful',
+                'fail': 'Regressions',
+                'unknown': 'Unknown',
+            },
+            size: {
+                height: 200,
+                width: 200,
+            },
+            radius: {inner: -30, outer: -42},
+        });
+    }
+
+    function listenForTableEvents(testCount) {
+        var btnList = ['total', 'pass', 'regressions', 'unknown'];
+
+        function _tableFilter(event) {
+            var activeId = event.target.id;
+            var status = activeId.substring('btn-'.length);
+
+            if (status == 'total') {
+                status = '';
+            } else if (status == 'regressions') {
+                status = 'fail';
+            }
+
+            gTestsTable.table.column(2).search(status).draw();
+
+            btnList.forEach(function(id) {
+                var btnId = 'btn-' + id;
+                var ele = document.getElementById(btnId);
+                console.log(btnId, activeId);
```

Comment:
> @gctucker are you using any linter? Maybe we could adopt one.

I'm not, probably because I'm not doing a lot of Javascript usually so I'm only using basic tools.  We could enable a linter with Travis CI though.

Reply:
Done - and yes probably, however they serve slightly different purposes.  Sometimes I want to keep printing some things in the console, it can be quicker as well.

Reply:
it makes sense.